state-chain/pallets/cf-ingress-egress/src/tests.rs

```diff
@@ -2120,14 +2122,175 @@ fn tainted_transactions_expire_if_not_witnessed() {
 		let tx_id = DepositDetails::default();
 		let expiry_at = System::block_number() + TAINTED_TX_EXPIRATION_BLOCKS as u64;
 
-		TaintedTransactions::<Test>::insert(BROKER, tx_id.clone(), expiry_at);
+		let (_, address) = request_address_and_deposit(BROKER, eth::Asset::Eth);
+		let _ = DepositChannelLookup::<Test, ()>::get(address).unwrap();
+
+		assert_ok!(<MockAccountRoleRegistry as AccountRoleRegistry<Test>>::register_as_broker(
+			&BROKER,
+		));
+
+		assert_ok!(IngressEgress::mark_transaction_as_tainted_inner(
+			RuntimeOrigin::signed(BROKER),
+			Default::default(),
+		));
+
+		System::set_block_number(expiry_at);
 
 		IngressEgress::on_idle(expiry_at, Weight::MAX);
 
-		assert!(!TaintedTransactions::<Test, ()>::contains_key(BROKER, tx_id.clone()));
+		assert!(!TaintedTransactions::<Test, ()>::contains_key(BROKER, tx_id));
 
 		assert_has_event::<Test>(RuntimeEvent::IngressEgress(
-			crate::Event::TaintedTransactionExpired { account_id: BROKER, tx_id },
+			crate::Event::TaintedTransactionExpired {
+				account_id: BROKER,
+				tx_id: Default::default(),
+			},
+		));
+	});
+}
+
+fn setup_boost_swap() -> ForeignChainAddress {
+	assert_ok!(
+		<MockAccountRoleRegistry as AccountRoleRegistry<Test>>::register_as_liquidity_provider(
+			&ALICE,
+		)
+	);
+
+	assert_ok!(IngressEgress::create_boost_pools(
+		RuntimeOrigin::root(),
+		vec![BoostPoolId { asset: eth::Asset::Eth, tier: 10 }],
+	));
+
+	<Test as crate::Config>::Balance::try_credit_account(&ALICE, eth::Asset::Eth.into(), 1000)
+		.unwrap();
+
+	let (_, address, _, _) = IngressEgress::request_swap_deposit_address(
+		eth::Asset::Eth,
+		eth::Asset::Eth.into(),
+		ForeignChainAddress::Eth(Default::default()),
+		Beneficiaries::new(),
+		BROKER,
+		None,
+		10,
+		None,
+		None,
+	)
+	.unwrap();
+
+	assert_ok!(IngressEgress::add_boost_funds(
+		RuntimeOrigin::signed(ALICE),
+		eth::Asset::Eth,
+		1000,
+		10
+	));
+
+	address
+}
+
+#[test]
+fn ignore_boosted_channels_if_tainted_after_prewitness() {
```

Comment:
This name seems misleading. What seems to be ingnored is the "tainted" status, but the name suggests that the channel (which I would read as "deposit") will be ignored, which is the opposite of what should happen.